xc7/utils/prjxray_create_edges.py

```diff
@@ -1031,39 +1046,45 @@ def main():
 
         edge_set = set()
 
-        for loc in progressbar.progressbar(grid.tile_locations()):
-            gridinfo = grid.gridinfo_at_loc(loc)
-            tile_name = grid.tilename_at_loc(loc)
+        c2 = conn.cursor()
+        for tile_pkey, tile_type_pkey, grid_x, grid_y in \
+                progressbar.progressbar(
+                c.execute(
+                    "SELECT pkey, tile_type_pkey, grid_x, grid_y FROM tile")):
+            loc = (grid_x, grid_y)
 
             # Not a synth node, check if in ROI.
-            if use_roi and not roi.tile_in_roi(loc):
```

Comment:
Why implement this comparison vs using the ROI object?  They should be equivilant?

Reply:
Why is this using VPR coordinates?  The ROI is defined using prjxray coordinates?  This should iterate over VPR tiles, and get the prjxray coordinates for the ROI check.

Reply:
That is another possible solution. However, the `prjxray_create_edges.py` does not get the roi directly. Instead it reads `synth_tiles.json` which contains the ROI range and synthetic tiles in VPR coordinates (already mapped). So it is easier to do a simple check whether a VPR location fits within the remapped ROI range read from `synth_tiles.json` rather than having the physical (prjxray) ROI definition as another input of the script. Both of them contain the same information.

Reply:
> Both of them contain the same information.

This is not generally speaking true.  The physical ROI limit has a 1:1 relationship to the underlying bitstream, the same does not have to be true for the remapped ROI.  Do not make the assumption that they are the same.

Reply:
What I mean is that `synth_tiles.json` contains the ROI which has been re-mapped from physical to VPR coordinates. So instead of passing the ROI from prjxray db to `prjxray_create_edges.py` (in physical coordinates) it is possible to read the roi in VPR coordinates (this is already done through reading `synth_tiles.json`) and re-map it back to physical coordinates. The check can be performed then. I never meant to assume 1:1 relation.

Reply:
@litghost I can add a "--roi" argument if you want and use it to pass the physical ROI from the prjxray. But it will only complicate the whole flow (which is in my opinion complex enough).

Reply:
@litghost Another solution that comes into my mind is to simply store physical ROI range in `synth_tiles.json`.

Reply:
That was what I was suggesting